src/Microsoft.AspNetCore.Http.Extensions/RequestHeaders.cs

```diff
@@ -249,6 +249,23 @@ public RangeConditionHeaderValue IfRange
             }
         }
 
+        public Uri Origin
```

Comment:
The spec format is a list. https://tools.ietf.org/html/rfc6454#section-7, and I'm not sure how you'd handle the 'null' case.

Reply:
Oh this is a problem, not only the 'null' case but apparently there can be multiple origins too.

> In some cases, a number of origins contribute to causing the user
   agents to issue an HTTP request.  In those cases, the user agent MAY
   list all the origins in the Origin header field.  For example, if the
   HTTP request was initially issued by one origin but then later
   redirected by another origin, the user agent MAY inform the server
   that two origins were involved in causing the user agent to issue the
   request.

Reply:
Should I remove the addition of the Origin stuff and keep this commit to only have the Referer stuff?

Reply:
Sure

Reply:
Okay, I removed the Origin part.